forc-test/src/lib.rs

```diff
@@ -204,6 +206,16 @@ impl<'a> PackageTests {
                 let (state, duration, receipts) =
                     exec_test(&pkg_with_tests.bytecode, offset, test_setup);
 
+                let gas_used = *receipts
+                    .iter()
+                    .find_map(|receipt| match receipt {
+                        tx::Receipt::ScriptResult { gas_used, .. } => Some(gas_used),
+                        _ => None,
+                    })
+                    .ok_or_else(|| {
+                        anyhow::anyhow!("missing used gas information from test execution")
+                    })?;
```

Comment:
Do we know what cases under which we can get a receipt that has no `gas_used` field?

Reply:
Hmm I looked for this and it is not stated [in the specs](https://fuellabs.github.io/fuel-specs/master/protocol/abi/receipts.html#scriptresult-receipt). As far as i understood from the specs and quick search, every script will return a script result that contains gas used. I tested also tested this with `should_revert` tests. Since each of our tests runs as scripts we should be able to get it all the time.